pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -1300,6 +1335,10 @@ type RetryStrategy struct {
 
 	// Backoff is a backoff strategy
 	Backoff *Backoff `json:"backoff,omitempty" protobuf:"bytes,3,opt,name=backoff,casttype=Backoff"`
+
+	// If exists, ScheduleOnDifferentHostNodes prevents of running workflow step on the same host
+	// Step won't rerun on the host where label is equal scheduleOnDifferentHostNodesLabel, e.g. "kubernetes.io/hostname" or "k3s.io/hostname"
+	ScheduleOnDifferentHostNodesLabel *string `json:"scheduleOnDifferentHostNodesLabel,omitempty" protobuf:"bytes,4,opt,name=scheduleOnDifferentHostNodesLabel"`
```

Comment:
if empty string is not a valid value, then you can make this simpler using `string` instead of `*string`

Reply:
That's changed too